### ISSUE TITLE
Backtrace generated when adding 3rd party apps to tendenci

### ISSUE THREAD

**Issue by goetzk** [tendenci/tendenci#468]
Hi,
I've just tried to add a simple app to my tendenci site to add a wiki (the purpose wasn't important, the learning was).
I've edited conf/local_urls.py to add the wiki to my list.

```
extrapatterns = patterns('',
    (r'^wiki/', include('waliki.urls')),
#     (r'^accounts/', include('allauth.urls')),
[more tendenci provided includes here]
```

What I was expecting was that tendenci would leave that namespace alone, but what happens is tendenci tries to do something (i haven't figured out what exactly) resulting in an error when i visit http://192.168.56.101/wiki/

```
NoReverseMatch at /wiki/
Reverse for 'category.update' with arguments '('', '', 1)' and keyword arguments '{}' not found. 1 pattern(s) tried: ['categories/update/(?P<app_label>\\w+)/(?P<model>\\w+)/(?P<pk>[\\w\\d]+)/$']
Request Method: GET
Request URL:    http://192.168.56.101/wiki/
Django Version: 1.8.4
Exception Type: NoReverseMatch
Exception Value:    
Reverse for 'category.update' with arguments '('', '', 1)' and keyword arguments '{}' not found. 1 pattern(s) tried: ['categories/update/(?P<app_label>\\w+)/(?P<model>\\w+)/(?P<pk>[\\w\\d]+)/$']
Exception Location: /srv/tendenci/venv/local/lib/python2.7/site-packages/django/core/urlresolvers.py in _reverse_with_prefix, line 496
Python Executable:  /srv/tendenci/venv/bin/python
Python Version: 2.7.9
Python Path:    
['/srv/tendenci/website',
 '/srv/tendenci/website',
 '/srv/tendenci/venv/bin',
 '/srv/tendenci/venv/lib/python2.7',
 '/srv/tendenci/venv/lib/python2.7/plat-x86_64-linux-gnu',
 '/srv/tendenci/venv/lib/python2.7/lib-tk',
 '/srv/tendenci/venv/lib/python2.7/lib-old',
 '/srv/tendenci/venv/lib/python2.7/lib-dynload',
 '/usr/lib/python2.7',
 '/usr/lib/python2.7/plat-x86_64-linux-gnu',
 '/usr/lib/python2.7/lib-tk',
 '/srv/tendenci/venv/local/lib/python2.7/site-packages']
Server time:    Tue, 1 Dec 2015 18:59:30 +1100
Error during template rendering

In template /srv/tendenci/venv/local/lib/python2.7/site-packages/tendenci/apps/pages/templates/pages/top_nav_items.html, error at line 93
Reverse for 'category.update' with arguments '('', '', 1)' and keyword arguments '{}' not found. 1 pattern(s) tried: ['categories/update/(?P<app_label>\\w+)/(?P<model>\\w+)/(?P<pk>[\\w\\d]+)/$']
83                                  </li>
84  
85                                  <li class="content-item">
86                                      <span class="app-name">
87                                          <a href="{% url 'page.edit.meta' app_object.pk %}">{% trans "Edit Meta" %}</a>
88                                      </span>
89                                  </li>
90  
91                                  <li class="content-item">
92                                      <span class="app-name">
93  
                                              <a href="
      {% url 'category.update' app_object.opt_app_label page.opt_module_name app_object.pk %}
      ">{% trans "Edit Categories" %}</a>


94                                      </span>
95                                  </li>
96                              {% endif %}
97  
98                              {% if can_delete_page %}
99                                  <li class="content-item">
100                                     <span class="app-name">
101                                         <a class="delete" href="{% url 'page.delete' app_object.pk %}">{% trans "Delete Page" %}</a>
102                                     </span>
103                                 </li>
Traceback Switch to copy-and-paste view

/srv/tendenci/venv/local/lib/python2.7/site-packages/django/core/handlers/base.py in get_response
                                response = wrapped_callback(request, *callback_args, **callback_kwargs) 
[remaining backtrace cut]
```

To confuse me more visiting http://192.168.56.101/wiki/page works in as much as there is no error but the page has the usual tendenci admin bar on top, sidebar on the right and footer at the bottom - until i click the 'add page' button when I appear to get the same error as when i loaded up /wiki/

```
NoReverseMatch at /wiki/page/edit
Reverse for 'category.update' with arguments '('', '', 3)' and keyword arguments '{}' not found. 1 pattern(s) tried: ['categories/update/(?P<app_label>\\w+)/(?P<model>\\w+)/(?P<pk>[\\w\\d]+)/$']
Request Method: GET
Request URL:    http://192.168.56.101/wiki/page/edit
Django Version: 1.8.4
Exception Type: NoReverseMatch
Exception Value:    
Reverse for 'category.update' with arguments '('', '', 3)' and keyword arguments '{}' not found. 1 pattern(s) tried: ['categories/update/(?P<app_label>\\w+)/(?P<model>\\w+)/(?P<pk>[\\w\\d]+)/$']
Exception Location: /srv/tendenci/venv/local/lib/python2.7/site-packages/django/core/urlresolvers.py in _reverse_with_prefix, line 496
Python Executable:  /srv/tendenci/venv/bin/python
Python Version: 2.7.9
Python Path:    
['/srv/tendenci/website',
 '/srv/tendenci/website',
 '/srv/tendenci/venv/bin',
 '/srv/tendenci/venv/lib/python2.7',
 '/srv/tendenci/venv/lib/python2.7/plat-x86_64-linux-gnu',
 '/srv/tendenci/venv/lib/python2.7/lib-tk',
 '/srv/tendenci/venv/lib/python2.7/lib-old',
 '/srv/tendenci/venv/lib/python2.7/lib-dynload',
 '/usr/lib/python2.7',
 '/usr/lib/python2.7/plat-x86_64-linux-gnu',
 '/usr/lib/python2.7/lib-tk',
 '/srv/tendenci/venv/local/lib/python2.7/site-packages']
Server time:    Tue, 1 Dec 2015 19:02:03 +1100
Error during template rendering

In template /srv/tendenci/venv/local/lib/python2.7/site-packages/tendenci/apps/pages/templates/pages/top_nav_items.html, error at line 93
Reverse for 'category.update' with arguments '('', '', 3)' and keyword arguments '{}' not found. 1 pattern(s) tried: ['categories/update/(?P<app_label>\\w+)/(?P<model>\\w+)/(?P<pk>[\\w\\d]+)/$']
83                                  </li>
84  
85                                  <li class="content-item">
86                                      <span class="app-name">
87                                          <a href="{% url 'page.edit.meta' app_object.pk %}">{% trans "Edit Meta" %}</a>
88                                      </span>
89                                  </li>
90  
91                                  <li class="content-item">
92                                      <span class="app-name">
93  
                                              <a href="
      {% url 'category.update' app_object.opt_app_label page.opt_module_name app_object.pk %}
      ">{% trans "Edit Categories" %}</a>


94                                      </span>
95                                  </li>
96                              {% endif %}
97  
98                              {% if can_delete_page %}
99                                  <li class="content-item">
100                                     <span class="app-name">
101                                         <a class="delete" href="{% url 'page.delete' app_object.pk %}">{% trans "Delete Page" %}</a>
102                                     </span>
103                                 </li>
Traceback Switch to copy-and-paste view

/srv/tendenci/venv/local/lib/python2.7/site-packages/django/core/handlers/base.py in get_response
                                response = wrapped_callback(request, *callback_args, **callback_kwargs) ...
```

Have i misunderstood the intended behaviour or is this a bug of some kind? If its expected behaviour could you provide some clues to how I should be fixing it?

thanks,

**Comment by goetzk** [comment]
I've done more testing (installed on a different system) and there have been no errors. everything has worked as expected and a page has been created in my wiki. must be something to do with my original install - i will try and determine what it is. i think that instance is still t7 so i will upgrade to t7.1 as a starter.

**Comment by eschipul** [comment]
that is simply saying it can't find your urls.py file in the installed apps in either /conf/settings.py or /local_settings.py

In this situation it might be easier to do "ls -a" and as a sudo cp that folder over, delete the existing contents and then put the wiki app in there.

zOMG as I type this I realize that is ridiculous. It's all I've got, but ridiculous. Need to add a new issue on best practices to install A standalone app like we did with /forums - in the past we were hooked on .env and json files which I now believe to not be a best practice. In this instance "ls -aH" may be your best friend.

Lastly we used to have a wiki but I had to remove it because of seo spammers. Maybe captcha will help with that now. Or security groups (oh ya, we just added security group limitations to forums!)
